File: 0002-add-two-numbers/0002-add-two-numbers.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        answer = ListNode(0)
        current = answer
        carry = 0

        while l1 or l2:
          x = l1.val if l1 else 0
          y = l2 .val if l2 else 0
          _sum = x + y + carry
          carry = _sum // 10

          current.next = ListNode(_sum % 10)   
          current = current.next
          if l1:
            l1 = l1.next
          if l2:
            l2 = l2.next
        
        if carry != 0:
          current.next = ListNode(1)

        return answer.next

File: 0002-add-two-numbers/0002-add-two-numbers.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func addTwoNumbers(l1 *ListNode, l2 *ListNode) *ListNode {
   answer := &ListNode{}
   current := answer
   carry := 0
    for l1 != nil || l2 != nil {
      x := 0
      if l1 != nil {
        x = l1.Val
        l1 = l1.Next
      }

      y := 0
      if l2 != nil {
        y = l2.Val
        l2 = l2.Next
      }
      sum := carry + x + y
      carry = sum / 10

      current.Next = &ListNode{Val : sum % 10}
      current = current.Next
    }

    if carry != 0 {
      current.Next = &ListNode{Val : 1}
    }
    return answer.Next
}